packages/service-integration/src/service_integration/pytest_plugin/docker_integration.py

```diff
@@ -0,0 +1,301 @@
+        path = tmp_dir / folder
+        if path.exists():
+            shutil.rmtree(path)
+        path.mkdir()
+        # we need to ensure the path is writable for the docker container (Gitlab-CI case)
+        os.chmod(str(path), 0o775)
+        assert path.exists()
+        host_folders[folder] = path
+
+    return host_folders
+
+
+@pytest.fixture
+def container_variables() -> Dict:
+    # of type INPUT_FOLDER=/home/scu/data/input
+    env = {
+        "{}_FOLDER".format(str(folder).upper()): (_CONTAINER_FOLDER / folder).as_posix()
+        for folder in _FOLDER_NAMES
+    }
+    return env
+
+
+@pytest.fixture
+def validation_folders(validation_dir: Path) -> Dict:
+    return {folder: (validation_dir / folder) for folder in _FOLDER_NAMES}
+
+
+@pytest.fixture
+def docker_container(
+    validation_folders: Dict,
+    host_folders: Dict,
+    docker_client: docker.DockerClient,
+    docker_image_key: str,
+    container_variables: Dict,
+) -> docker.models.containers.Container:
+    # copy files to input folder, copytree needs to not have the input folder around.
+    host_folders["input"].rmdir()
+    shutil.copytree(validation_folders["input"], host_folders["input"])
+    assert Path(host_folders["input"]).exists()
+    # run the container (this may take some time)
+    container = None
+    try:
+        volumes = {
+            host_folders[folder]: {
+                "bind": container_variables["{}_FOLDER".format(str(folder).upper())]
+            }
+            for folder in _FOLDER_NAMES
+        }
+        container = docker_client.containers.run(
+            docker_image_key,
+            "run",
+            detach=True,
+            remove=False,
+            volumes=volumes,
+            environment=container_variables,
+        )
+        response = container.wait()
+        if response["StatusCode"] > 0:
+            logs = container.logs(timestamps=True)
+            pytest.fail(
+                "The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(
+                    response["StatusCode"],
+                    "run",
+                    pformat(
+                        (container.logs(timestamps=True).decode("UTF-8")).split("\n"),
+                        width=200,
+                    ),
+                )
+            )
+        else:
+            yield container
+    except docker.errors.ContainerError as exc:
+        # the container did not run correctly
+        pytest.fail(
+            "The container stopped with exit code {}\n\n\ncommand:\n {}, \n\n\nlog:\n{}".format(
+                exc.exit_status,
+                exc.command,
+                pformat(
+                    (container.logs(timestamps=True).decode("UTF-8")).split("\n"),
+                    width=200,
+                )
+                if container
+                else "",
+            )
+        )
+    finally:
+        # cleanup
+        if container:
+            container.remove()
```

Comment:
this could fail, do we care or just let the error show.

Reply:
suppressed exceptions here. Note that this is for testing!